python/RESOURCE_MANAGED_BY_CLOUDFORMATION/RESOURCE_MANAGED_BY_CLOUDFORMATION.py

```diff
@@ -0,0 +1,495 @@
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+allResources = dict()
+
+
+# Check if key exists in dict
+def checkKey(dict, key):
+    if key in dict.keys():
+        return True
+    else:
+        return False
+
+
+def build_cloudformation_resource_list():
+    cfn = boto3.client('cloudformation')
+    stackList = cfn.list_stacks(
+        StackStatusFilter=[
+            'CREATE_COMPLETE',
+            'ROLLBACK_COMPLETE',
+            'UPDATE_COMPLETE',
+            'UPDATE_ROLLBACK_COMPLETE',
+        ]
+    )
+    try:
+        print(stackList['NextToken'])
+    except KeyError:
+        # print('no stack truncation')
+        pass
+        # print(stackList['StackSummaries'])
+    # NextToken handling
+    # while stackList.NextToken
+    for stack in stackList['StackSummaries']:
+        stackResources = cfn.list_stack_resources(
+            StackName=stack['StackId']
+        )
+        try:
+            print(stackResources['NextToken'])
+        except KeyError:
+            # print('no truncation')
+            pass
+        for resource in stackResources['StackResourceSummaries']:
+            if not checkKey(allResources, resource['ResourceType']):
+                # allResources[resource['ResourceType']] = []
+                allResources[resource['ResourceType']] = set()
+            try:
+                # allResources[resource['ResourceType']].append(resource['PhysicalResourceId'])
+                allResources[resource['ResourceType']].add(resource['PhysicalResourceId'])
+            except KeyError:
+                continue  # PhysicalResourceId is not available
+
+
+def build_iam_role_list():
+    roleList = []
+    iam = boto3.client('iam')
+    # print(iam.list_roles()['IsTruncated'])
+    for role in iam.list_roles()['Roles']:
+        # exceptions
+        if not role['Path'].startswith('/aws-service-role/'):
+            roleList.append(role)
+    return roleList
+
+
+def build_iam_managed_policy_list():
+    policyList = []
+    iam = boto3.client('iam')
+    # print(iam.list_policies()['IsTruncated'])\
+    policyListResult = iam.list_policies(
```

Comment:
Query config instead, it will less hammer iam apis

Reply:
I had purposely made a design choice to query the service API directly to capture any resource types which do not have Config support yet. For example, `AWS::IAM::ManagedPolicy` resources are not supported by `config_client.list_discovered_resources` and are not listed as supported types in the [documentation](https://docs.aws.amazon.com/config/latest/developerguide/resource-config-reference.html).

Let me know if you have a better way of handling this.